framework/helpers/BaseHtml.php

```diff
@@ -1693,11 +1693,18 @@ public static function renderSelectOptions($selection, $items, &$tagOptions = []
         $encodeSpaces = ArrayHelper::remove($tagOptions, 'encodeSpaces', false);
         $encode = ArrayHelper::remove($tagOptions, 'encode', true);
         if (isset($tagOptions['prompt'])) {
-            $prompt = $encode ? static::encode($tagOptions['prompt']) : $tagOptions['prompt'];
+            $attrs = ['value' => ''];
+            if (is_string($tagOptions['prompt'])) {
+                $text = $tagOptions['prompt'];
+            } else {
+                $text = $tagOptions['prompt']['text'];
```

Comment:
I haven't added PHPDoc yet because I was not sure about implementation (that's why I listed different options in initial post). If it's OK I will add it.